server/src/main/java/io/deephaven/server/table/ops/TableServiceGrpcImpl.java

```diff
@@ -296,12 +305,116 @@ public void updateBy(UpdateByRequest request, StreamObserver<ExportedTableCreati
+                return doubleValue.intValue();
+            }
+            if (dataType == long.class) {
+                return doubleValue.longValue();
+            }
+            if (dataType == float.class) {
+                return doubleValue.floatValue();
+            }
+            if (dataType == double.class) {
+                return doubleValue;
+            }
+            if (dataType == BigInteger.class) {
+                return BigDecimal.valueOf(doubleValue).toBigInteger();
+            }
+            if (dataType == BigDecimal.class) {
+                return BigDecimal.valueOf(doubleValue);
+            }
+        } else if (literal.hasBoolValue()) {
+            return literal.getBoolValue();
+        }
+        throw GrpcUtil.statusRuntimeException(Code.INVALID_ARGUMENT, "Invalid column type for seek: " + dataType);
+    }
+
     @Override
     public void whereIn(WhereInRequest request, StreamObserver<ExportedTableCreationResponse> responseObserver) {
         oneShotOperationWrapper(BatchTableRequest.Operation.OpCase.WHERE_IN, request, responseObserver);
     }
 
     @Override
+    public void seekRow(SeekRowRequest request, StreamObserver<SeekRowResponse> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            final SessionState session = sessionService.getCurrentSession();
+            final Ticket sourceId = request.getSourceId();
+            if (sourceId.getTicket().isEmpty()) {
+                throw GrpcUtil.statusRuntimeException(Code.FAILED_PRECONDITION, "No consoleId supplied");
+            }
+
+            SessionState.ExportObject<Table> exportedTable =
+                    ticketRouter.resolve(session, sourceId, "sourceId");
+            session.nonExport()
+                    .require(exportedTable)
+                    .onError(responseObserver)
+                    .submit(() -> {
+                        final Table table = exportedTable.get();
+                        final String columnName = request.getColumnName();
+                        final Class<?> dataType = table.getDefinition().getColumn(columnName).getDataType();
+                        final Object seekValue = getSeekValue(request.getSeekValue(), dataType);
+                        final Long result = table.apply(new SeekRow(
+                                request.getStartingRow(),
+                                columnName,
+                                seekValue,
+                                request.getInsensitive(),
+                                request.getContains(),
+                                request.getIsBackward()));
+                        safelyExecuteLocked(responseObserver, () -> {
+                            SeekRowResponse.Builder rowResponse = SeekRowResponse.newBuilder();
+                            responseObserver.onNext(rowResponse.setResultRow(result).build());
+                            responseObserver.onCompleted();
+                        });
```

Comment:
Safely execute is gone now, with a new call to use instead that will prevent accidentally creating the response inside the "safe" part. See GrpcUtil.safelyComplete. Roughly, this will be:

```suggestion
                        SeekRowResponse.Builder rowResponse = SeekRowResponse.newBuilder();
                        safelyComplete(responseObserver, rowResponse.setResultRow(result).build());
```